CMake/Version.cmake

```diff
@@ -0,0 +1,72 @@
+
+# Try to figure out  versioning scheme for dynamic libraries
+# based on the system we're compiling for
+function(determine_version_type _out_version_type_name)
+
+        set(_ver_type "none")
+
+        if(WIN32 OR CYGWIN)
+                set(_ver_type "windows")
+        else()
+                # this part tries to mimic autotools as close as possible
+                if(CMAKE_SYSTEM_NAME STREQUAL "AIX" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "Linux" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "Haiku" OR
+                   CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+                        set(_ver_type "linux")
```

Comment:
Ok, so SunOS is type "linux" ...

Reply:
Here I have a big problem. Autotools have complex rules for changing the output of ```uname``` into some more detailed representation and then derive those version types from it. It has separate rules for Solaris and SunOS. Unfortunately both present themselves as "SunOS". I've chosen to assume Solaris here which is linux-type.